osu.Server.Queues.ScoreStatisticsProcessor/DatabaseHelper.cs

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Threading.Tasks;
+using Dapper;
+using Dapper.Contrib.Extensions;
+using MySqlConnector;
+using osu.Server.Queues.ScoreStatisticsProcessor.Models;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor
+{
+    public static class DatabaseHelper
+    {
+        /// <summary>
+        /// Retrieve user stats for a user based on a score context.
+        /// Creates a new entry if the user does not yet have one.
+        /// </summary>
+        /// <param name="score">The score to use for the user and ruleset lookup.</param>
+        /// <param name="db">The database connection.</param>
+        /// <param name="transaction">The database transaction, if one exists.</param>
+        /// <returns>The retrieved user stats. Null if the ruleset or user was not valid.</returns>
+        public static UserStats? GetUserStats(SoloScoreInfo score, MySqlConnection db, MySqlTransaction? transaction = null)
```

Comment:
Do we really need this dual pathing? I'd prefer to async everything as discussed.

Reply:
Hmm, this changed to be non-async only. Was there a reason for choosing that?

Reply:
I thought doing it this way basically serves as an extraction of the existing methods from `StatisticsProcessor`. I can use the async forms instead if you'd rather.

Reply:
Just thinking that's the direction we want to go, if any.